packages/experimental/kai/src/playwright/app-manager.ts

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import type { Context as MochaContext } from 'mocha';
+import type { ConsoleMessage, Page } from 'playwright';
+
+import { sleep, synchronized, Trigger } from '@dxos/async';
+import { setupPage } from '@dxos/test';
+
+// TODO(wittjosiah): Get this from executor.
+const BASE_URL = 'http://localhost:4200';
+
+export class AppManager {
+  page!: Page;
+
+  private _initialized = false;
+  private _invitationCode = new Trigger<string>();
+  private _authenticationCode = new Trigger<string>();
+
+  constructor(private readonly mochaContext: MochaContext) {}
+
+  // Getters
+
+  async currentSpace() {
+    const url = await this.page.url();
+    return url.split('/').find((part) => part.includes('...'));
+  }
+
+  async stackIsVisible() {
+    await this._init();
+
+    return (
+      (await this.page.isVisible('a:has-text("Stack")')) &&
```

Comment:
i think this kind of testing is too brittle.
what do you think about only testing for `data-xxx` elements?

Reply:
this is just placeholder so that there's a test at all, see todomvc app manager for a more fleshed out example https://github.com/dxos/dxos/blob/main/packages/apps/todomvc/src/playwright/app-manager.ts